.circleci/config.yml

```diff
@@ -179,103 +201,123 @@ jobs:
-      - run:
-          command: |
-            eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v t/api/*.t
+  t.ui:
+    <<: *test-template
+    environment:
+      TEST_PART=ui
+
+  t.unstable:
+    <<: *test-template
+    environment:
+      TEST_PART=unstable
+
+  t.api:
+    <<: *test-template
+    environment:
+      TEST_PART=api
+
+  t.developer:
+    <<: *test-fullstack-template
+    environment:
+      TEST_PART=developer-fullstack
+
+  t.scheduler:
+    <<: *test-fullstack-template
+    environment:
+      TEST_PART=scheduler-fullstack
 
-  testfullstack:
+  report.codecov:
     docker:
       - <<: *base
     steps:
       - checkout
-      - run: echo CIRCLE_WORKFLOW_ID=$CIRCLE_WORKFLOW_ID
-      - run: *chown_hack_for_cache_integration
-      - restore_cache: *restore_cache
-      - restore_cache: *restore_integration_cache
-      - run: *check_cache
-      - run: *install_cached_packages
-      - run: *build_autoinst
-      - run:
-          command: |
-            eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib FULLSTACK=1 prove -v t/full-stack.t
-
-  testdeveloperfullstack:
+      - attach_workspace:
+          at: .
+      - run: sudo zypper ar -f https://download.opensuse.org/repositories/devel:/languages:/perl/openSUSE_Leap_15.1/ devel-perl
+      - run: sudo zypper --gpg-auto-import-keys ref
+      - run: sudo zypper -n install --from devel-perl perl-Devel-Cover perl-Devel-Cover-Report-Codecov # 1.29 cannot merge DB
+      - run: make coverage-merge-db
+      - run: cover -report html_minimal
+      - run: cover -report codecov
+  
+  build.docs:
     docker:
       - <<: *base
+    environment:
+      COMMIT_AUTHOR_EMAIL=skynet@open.qa
     steps:
       - checkout
-      - run: echo CIRCLE_WORKFLOW_ID=$CIRCLE_WORKFLOW_ID
-      - run: *chown_hack_for_cache_integration
-      - restore_cache: *restore_cache
-      - restore_cache: *restore_integration_cache
-      - run: *check_cache
-      - run: *install_cached_packages
-      - run: *build_autoinst
       - run:
           command: |
-            eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib DEVELOPER_FULLSTACK=1 prove -v t/33-developer_mode.t
-
-  testschedulerfullstack:
-    docker:
-      - <<: *base
-    steps:
-      - checkout
-      - run: echo CIRCLE_WORKFLOW_ID=$CIRCLE_WORKFLOW_ID
-      - run: *chown_hack_for_cache_integration
-      - restore_cache: *restore_cache
-      - restore_cache: *restore_integration_cache
-      - run: *check_cache
-      - run: *install_cached_packages
-      - run: *build_autoinst
+            sudo zypper ar -f https://download.opensuse.org/repositories/devel:/languages:/perl/openSUSE_Leap_15.1/ devel-perl
+            sudo zypper ar -f https://download.opensuse.org/repositories/devel:/languages:/ruby/openSUSE_Leap_15.1/ devel-ruby
+            sudo zypper ar -f https://download.opensuse.org/repositories/devel:/languages:/ruby:/extensions/openSUSE_Leap_15.1/ devel-ruby-extensions
```

Comment:
see above

Reply:
This is a bit different and I don't want to force dependencies for documentation generation on regular development and testing

Reply:
hm, ok. Why did we not need this in before? Could it be that the Leap 42.3 based devel:openQA repo has these dependencies but not yet for Leap 15.1?

Reply:
They were installed from CI as well https://github.com/os-autoinst/openQA/blob/bb62d7666521acd8a0ff57e1b7c65d3f49ae9a55/script/generate-documentation#L152 . But current PR leaves old travis doc generation in place and adds own docs generation into openqabot/openQA, so we can preview results generated by circleci (on 15.1) and switch it over in another PR

Reply:
Currently "generate-documentation" installs asciidoctor deps from ruby gems, if this is what you mean. So why do you add yet another way here then?

Reply:
I actually prefer using zypper. Adding additional repos for that is less "different" than using a different package manager. So this makes it actually more consistent.

Reply:
> So why do you add yet another way here then?

I just did follow what the script was pointing me to do https://github.com/os-autoinst/openQA/blob/882496f0b34f1c87286d4a3cca9d7f1ba91e7a9f/script/generate-documentation#L110 . I am failing to understand what is problem here? You are feel free to make it simpler afterward, but why is it blocking now?

Reply:
I think I am interested just as you are in getting work done fast and not block something. What I see here is that in a PR – which has now the very unspecific title "Rework CI" – you are doing a lot of things which are all going in the right direction but should be handled as individual changes. You can of course work on these many things in one PR but I recommend for easier review – and faster acceptance and merging – to focus on individual topics at a time in individual PRs. I agree of course with using RPM packages for this cause over ecosystem custom packaging. So far our approach was to provide all additional dependencies for building and testing openQA within devel:openQA and its subprojects by linking the according packages from Factory there. If there are dependencies missing please make sure they are submitted to openSUSE:Factory first and then linked in devel:openQA.

Reply:
I'd also prefer it these packages would be in our devel:openQA subproject (instead of adding other devel repos directly). But it sounds a bit overelaborate to separate these kind of changes from this PR.

Reply:
I'm just slightly confused. The `generate-documentation` script already runs these commands, right? It does for Travis. Why not for CircleCI?

Reply:
ie. the `install_asciidoctor()` function is meant to achieve this.

Reply:
exactly my point. It will probably work to add the additional repos here but it's duplicating without cleaning up the according code in generate-documentation. Apparently @mkittler and @andrii-suse still think this is more effort to separate into another PR whereas my opinion differs in this regard.

Reply:
I think you removed that part again so I guess this conversation can be resolved.